gobblin-runtime/src/main/java/gobblin/runtime/AbstractJobLauncher.java

```diff
@@ -623,61 +625,42 @@ private EventSubmitter buildEventSubmitter(List<? extends Tag<?>> tags) {
    * @param taskExecutor a {@link TaskExecutor} for task execution
    * @param taskStateStore a {@link StateStore} for storing {@link TaskState}s
    * @param logger a {@link Logger} for logging
+   * @param shouldCommitTasks whether should commit {@link GobblinMultiTaskAttempt}
```

Comment:
I think this parameter name and javadoc are misleading. Ultimately, all successful tasks should commit. The questions is whether they should commit immediately after each run or at the end. Please think about improving the naming. Using a enum instead of boolean can also improve readability.